lib/Cargo.toml

```diff
@@ -44,6 +44,9 @@ either = { workspace = true }
 futures = { workspace = true }
 git2 = { workspace = true, optional = true }
 gix = { workspace = true, optional = true }
+gix-actor = { workspace = true }
+gix-date = { workspace = true }
```

Comment:
nit: do we need to declare these two? gix-mailmap seems to depend on them, so it should be enough to declare only that dependency, no?

Reply:
Maybe feature flag of `gix` can be set conditionally instead?

Reply:
We need `gix_actor::SignatureRef` in `jj_lib::mailmap`, so that needs to be an explicit dependency. `gix-date` was formerly required but no longer is; I’ve removed it, thank you!

The `gix` crate has [a lot of dependencies](https://github.com/Byron/gitoxide/blob/8d89b865c84d1fb153d93343d1ce4e1d64e53541/gix/Cargo.toml#L293) even with all features disabled. They might fall out during linking, but I figured that it would be best to avoid bringing more of Gitoxide into the no‐Git‐backend build than necessary (it’s a concern for compile times too, after all), and that the explicit use of `git_{mailmap,actor}::*` would signal a distinction of these utility libraries from the `gix::*` used solely for the Git backend. However, if people would prefer to always bring in `gix` and just conditionalize features on it I’m happy to switch.

Reply:
I'd recommend to stick to plumbing as long as there is no disadvantage. With Mailmaps ,the involved thing is the initial setup in case you want to support all the git configuration around mailmaps as well. For that, it would be easiest to use `gix::Repository::*mailmap*()` to obtain a ready-made one.
*Note that I am probably missing a lot of context, as I only searched for `gix`, apologies if this is more noise than signal.*

Reply:
We use the plumbing crate for interfacing with our Git backend, but in this case we’re using it purely for handling the `.mailmap` format directly, even when the Git backend is disabled, so there’s no Git repository lookup going on.

Reply:
I see, thanks for the clarification - in this case reading a single `.mailmap` with the plumbing crate will be fine.
And: yay to the modular crate approach :).

Reply:
I’ve now dropped the unused `gix-date` dependency.